electricitymap/contrib/capacity_parsers/GB.py

```diff
@@ -0,0 +1,58 @@
+from datetime import datetime
+from logging import getLogger
+from typing import Dict, Union
+
+from bs4 import BeautifulSoup
+from requests import Response, Session
+
+from electricitymap.contrib.config import ZoneKey
+
+logger = getLogger(__name__)
+MODE_MAPPING = {
+    '"Wind Onshore"': "wind",
+    '"Wind Offshore"': "wind",
+    '"Solar"': "solar",
+    '"Other renewable"': "unknown",
+    '"Other"': "unknown",
+    '"Nuclear"': "nuclear",
+    '"Hydro Run-of-river and poundage"': "hydro",
+    '"Fossil Hard coal"': "coal",
+    '"Fossil Gas"': "gas",
+    '"Biomass"': "biomass",
+    '"Hydro Pumped Storage"': "hydro storage",
+}
+
+SOURCE = "bmreports.com"
+
+
+def fetch_production_capacity(
+    zone_key: ZoneKey, target_datetime: datetime, session: Session
+) -> Union[Dict, None]:
+    url = f"https://www.bmreports.com/bmrs/?q=ajax/year/B1410/{target_datetime.year}/"
+    r: Response = session.get(url)
+
+    if r.status_code == 200:
```

Comment:
I would recommend inverting this.
```suggestion
    if not r.ok:
       raise
```

Reply:
The reason is that you will nest the rest of the code one level less and it is then more readable.